### PR TITLE
Add SentryOptions.AutoRegisterTracing

### PR DESCRIPTION
**Pull request by jamescrosswell**
Resolves https://github.com/getsentry/sentry-dotnet/issues/2865

Needs [corresponding documentation](https://github.com/getsentry/sentry-docs/issues/8532#issuecomment-1819755354) when released.

### CODE REVIEW

File: src/Sentry.AspNetCore/SentryAspNetCoreOptions.cs [hunk 1]
```diff
@@ -65,6 +66,21 @@ public class SentryAspNetCoreOptions : SentryLoggingOptions
...
     public bool AdjustStandardEnvironmentNameCasing { get; set; } = true;
 
+    /// <summary>
+    /// <para>
+    /// When true (by default) Sentry automatically registers its tracing middleware immediately after
+    /// `EndpointRoutingApplicationBuilderExtensions.UseRouting`.
+    /// </para>
+    /// <para>
+    /// If you need to control when Sentry's tracing middleware is registered, you can set
+    /// <see cref="AutoRegisterTracing"/> to false call
```
Comment (bruno-garcia) [inline src/Sentry.AspNetCore/SentryAspNetCoreOptions.cs:76]:
```suggestion
    /// <see cref="AutoRegisterTracing"/> to false and call
```

File: src/Sentry.AspNetCore/SentryTracingMiddlewareExtensions.cs [hunk 2]
```diff
@@ -11,32 +14,30 @@ namespace Microsoft.AspNetCore.Builder;
 [EditorBrowsable(EditorBrowsableState.Never)]
 public static class SentryTracingMiddlewareExtensions
 {
+    internal const string AlreadyRegisteredWarning = "Sentry tracing middleware has already registered. This call to UseSentryTracing is unnecessary.";
```
Comment (bruno-garcia) [inline src/Sentry.AspNetCore/SentryTracingMiddlewareExtensions.cs:17]:
```suggestion
    internal const string AlreadyRegisteredWarning = "Sentry tracing middleware was already registered. This call to UseSentryTracing is unnecessary.";
```

**Review by bruno-garcia — APPROVED** [review]
Thanks for improving this!

Added what I thought made sense but ESL so maybe it was right originally

**Comment by bruno-garcia** [comment]
Windows CI failed twice in a row: [1](https://github.com/getsentry/sentry-dotnet/actions/runs/6940542623/attempts/1?pr=2871) and [2](https://github.com/getsentry/sentry-dotnet/actions/runs/6940542623/attempts/2?pr=2871)

**Comment by jamescrosswell** [comment]
> Windows CI failed twice in a row: [1](https://github.com/getsentry/sentry-dotnet/actions/runs/6940542623/attempts/1?pr=2871) and [2](https://github.com/getsentry/sentry-dotnet/actions/runs/6940542623/attempts/2?pr=2871)

Yeah it started failing on all the PRs for the main branch... but just before I was about to go to sleep so haven't had a chance to look at it yet. You can see neither of the changes in this PR caused it (they were your changes to code comments). This is a non-deterministic CI problem again... probably GitHub updated their windows runner or something.

**Pull request merged**